### PR TITLE
FIX: Rebuild shared library to fix linux - nvidia error

### PR DESCRIPTION
**Pull request by ripjackie**
When I posted in ChloePrime/AAAParticles#22, I did some digging into the issue to further find out what could be the cause. I discovered that the issue wasn't anything to do with OpenGL, Mesa, Nvidia, Etc. I believe it may have been an issue in how you built `libEffekseerNativeForJava.so`. Noting that the symbol `glGenSamplers` was undefined in the library and running `ldd` did not show any GL shared libs being pulled, I figured it was a build issue and it couldn't find OpenGL. I pulled the repo effekseer/EffekseerForMultiLanguages, pulled everything to the newest version, fixed a couple include issues, and rebuild the shared library. I replaced the old shared library in my minecraft folder with this new built one, and minecraft fired right up. 

In Summary:
I believe the shared library `libEffekseerNativeForJava.so` was built incorrectly.
I rebuilt it and replaced it, and minecraft fires right up.
The issue doesn't seem to be related to drivers at all, considering most computers support some form of OpenGL 3.3+.

If I can be of any further assistance, please let me know!

This so was built on:
Arch Linux x86_64 ( Linux-zen 6.10.10 )
nvidia 560.35.03

### CODE REVIEW

**Comment by ChloePrime** [comment]
image
LGTM

**Comment by ChloePrime** [comment]
> When I posted in #22, I did some digging into the issue to further find out what could be the cause. I discovered that the issue wasn't anything to do with OpenGL, Mesa, Nvidia, Etc. I believe it may have been an issue in how you built `libEffekseerNativeForJava.so`. Noting that the symbol `glGenSamplers` was undefined in the library and running `ldd` did not show any GL shared libs being pulled, I figured it was a build issue and it couldn't find OpenGL. I pulled the repo effekseer/EffekseerForMultiLanguages, pulled everything to the newest version, fixed a couple include issues, and rebuild the shared library. I replaced the old shared library in my minecraft folder with this new built one, and minecraft fired right up.
> 
> In Summary: I believe the shared library `libEffekseerNativeForJava.so` was built incorrectly. I rebuilt it and replaced it, and minecraft fires right up. The issue doesn't seem to be related to drivers at all, considering most computers support some form of OpenGL 3.3+.
> 
> If I can be of any further assistance, please let me know!
> 
> This so was built on: Arch Linux x86_64 ( Linux-zen 6.10.10 ) nvidia 560.35.03

btw can you help me fix this issule on Windows and fix #31 ?

**Comment by HugKitten** [comment]
> If I can be of any further assistance, please let me know!

Mind checking if [Eldritch_End](https://github.com/nvb-uy/Eldritch_End) has the same issue as well?

**Comment by ChloePrime** [comment]
> > If I can be of any further assistance, please let me know!
> 
> Mind checking if [Eldritch_End](https://github.com/nvb-uy/Eldritch_End) has the same issue as well?

Eldritch End jarjars this mod. So put new version of AAAParticles in mods folder may fix its issue

**Comment by HugKitten** [comment]
> > > If I can be of any further assistance, please let me know!
> > 
> > 
> > Mind checking if [Eldritch_End](https://github.com/nvb-uy/Eldritch_End) has the same issue as well?
> 
> Eldritch End jarjars this mod. So put new version of AAAParticles in mods folder may fix its issue

`Caused by: java.lang.UnsatisfiedLinkError: /home/sarah/.var/app/org.prismlauncher.PrismLauncher/data/PrismLauncher/instances/2/minecraft/libEffekseerNativeForJava.so: libGLU.so.1: cannot open shared object file: No such file or directory
	at java.base@21.0.4/jdk.internal.loader.NativeLibraries.load(Native Method)
	at java.base@21.0.4/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:331)
	at java.base@21.0.4/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:197)
	at java.base@21.0.4/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:139)
	at java.base@21.0.4/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2418)
	at java.base@21.0.4/java.lang.Runtime.load0(Runtime.java:852)
	at java.base@21.0.4/java.lang.System.load(System.java:2025)
	at mod.chloeprime.aaaparticles.client.AAAParticlesClient.installNativeLibrary(AAAParticlesClient.java:53)
	... 12 more`

**Comment by ripjackie** [comment]
@HugKitten, that looks like the shared lib is not finding java 21? strange, considering I built the lib using Java 17. To check, do you have java 21?

**Comment by HugKitten** [comment]
> @HugKitten, that looks like the shared lib is not finding java 21? strange, considering I built the lib using Java 17. To check, do you have java 21?

Yes, both 17 and 21.

**Comment by ripjackie** [comment]
@HugKitten Alright, now also double check that you have libGLU installed and updated to the lastest version? if you have java 21 then that's the only missing link. If that doesn't solve it, give me some sysinfo and I'll dig deeper.

**Comment by nvb-uy** [comment]
> > If I can be of any further assistance, please let me know!
> 
> Mind checking if [Eldritch_End](https://github.com/nvb-uy/Eldritch_End) has the same issue as well?

I'll jarjar the updated version in the next update. Is it confirmed that works fine on linux & mac now? I can't test personally

**Comment by ripjackie** [comment]
> > > If I can be of any further assistance, please let me know!
> > 
> > 
> > Mind checking if [Eldritch_End](https://github.com/nvb-uy/Eldritch_End) has the same issue as well?
> 
> I'll jarjar the updated version in the next update. Is it confirmed that works fine on linux & mac now? I can't test personally

Working fine for me on archlinux w/ kernel linux-zen 6.10.10, I'm a couple hours into a playthrough using stock Prominince 2 3.0.2 with only changing this mod version. No idea about mac
